src/utils/wallet.js

```diff
@@ -512,20 +934,37 @@ class Wallet {
     }
 
     async deleteRecoveryMethod({ kind, publicKey }) {
+        const res = await this.removeAccessKey(publicKey)
+        console.log('deleteRecoveryMethod', res)
         await this.postSignedJson('/account/deleteRecoveryMethod', {
             accountId: this.accountId,
             kind,
             publicKey
         })
-        await this.removeAccessKey(publicKey)
     }
 
-    async recoverAccountSeedPhrase(seedPhrase) {
+    async recoverAccountSeedPhrase(seedPhrase, use2fa = false, accountId, fromSeedPhraseRecovery = false) {
```

Comment:
I am not sure why this method needs so many changes. Shouldn't it just change the method it calls to add a key based on whether account has multisig contract?

Reply:
2fa and local access keys / seed phrase fak with 2fa introduces several new paths

Reply:
We've cleaned this up even more.